lib/tests/util/file.cc

```diff
@@ -1,76 +0,0 @@
-#include <catch.hpp>
```

Comment:
I went through and added some test cases based on these, mostly the ones that ensure a false callback will stop processing.